src/world.hpp

```diff
@@ -124,7 +126,9 @@ class World
 
 		int admin_count;
 
-		World(std::array<std::string, 6> dbinfo, const Config &eoserv_config, const Config &admin_config);
+		World(const std::shared_ptr<DatabaseFactory>& databaseFactory, const Config &eoserv_config, const Config &admin_config);
```

Comment:
Any reason databaseFactory is being passed as a ref to a `shared_ptr` rather than just passing the `shared_ptr` itself?

Reply:
Nah. I can change it. I think I was trying to use unique_ptr at first but it made it hard to test so I went with shared_ptr instead